tests/kernel/usleep/src/main.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019 Intel Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <zephyr.h>
+
+/*
+ * precision timing tests in an emulation environment are not reliable.
+ * if the test passes at least once, we know it works properly, so we
+ * attempt to repeat the test RETRIES times before reporting failure.
+ */
+
+#define RETRIES		1000
+
+/*
+ * test parameters. SLEEP_US * LOOPS should be at least a few milliseconds
+ * so that the precision of the system uptime clock (1ms) isn't a factor.
+ */
+
+#define SLEEP_US	50
+#define LOOPS		200
+
+#define LOWER_BOUND_MS	((SLEEP_US * LOOPS) / 1000)
+#define UPPER_BOUND_MS	(LOWER_BOUND_MS * 2)
+
+void test_usleep(void)
+{
+	int retries = 0;
+	s64_t elapsed_ms;
+
+	while (retries < RETRIES) {
+		s64_t start_ms;
+		s64_t end_ms;
+		int i;
+
+		++retries;
+
+		start_ms = k_uptime_get();
+
+		for (i = 0; i < LOOPS; ++i)
```

Comment:
Minor style issue here, I wonder why checkpatch does not complain about this. Anyway, you need to add {} to the one liners in this this function (here and couple of lines below).

Reply:
My understanding is that the Linux style guidelines (that checkpatch is meant to enforce) specifically decry the use of braces around simple one-liners:

(http://lxr.linux.no/linux+v3.6.1/Documentation/CodingStyle#L157)

But the braces are required by MISRA, right?

Reply:
I thought that checkpatch would have checked the braces but actually it does not do it.
And yes, it is some MISRA rule.